### PR TITLE
UAF-46 / UAF-2973 Procurement Card Reconciliation Step 10 (Notes and Attachments)

### PR DESCRIPTION
**Pull request by adamkost**
Note: The scenario for "Enter Sales Tax Amount" could not be properly tested. This field is added in UAF-2972, and was not available at this time for proper testing. Based on the code currently available, it *should* work. Once the code's been reviewed and merged, this may need to be revisited.

### CODE REVIEW

**Comment by ghost** [comment]
This branch now has conflicts that need to be resolved prior to code review.

File: kfs-core/src/main/resources/edu/arizona/kfs/fp/businessobject/datadictionary/ProcurementCardHolder.xml [hunk 1]
```diff
@@ -205,6 +205,9 @@
 		<property name="label" value="Card Holder Postal (ZIP) Code" />
 		<property name="shortLabel" value="Code" />
 		<property name="required" value="false"/>
+		<property name="validationPattern">
+			<bean parent="AnyCharacterValidationPattern"/>
+		</property>
```
Comment (nataliac06) [inline kfs-core/src/main/resources/edu/arizona/kfs/fp/businessobject/datadictionary/ProcurementCardHolder.xml:210]: Do we need to add the maxLenght for this field as I didn't see it in the parent xml:
<property name="maxLength" value="20" />

File: kfs-core/src/main/resources/edu/arizona/kfs/fp/businessobject/datadictionary/ProcurementCardTransactionDetail.xml [hunk 2]
```diff
@@ -106,6 +106,11 @@ http://rice.kuali.org/dd http://rice.kuali.org/dd/dd.xsd">
 </bean>
 
 <!-- Attribute Definitions -->
+    <bean id="ProcurementCardTransactionDetail-transactionOriginalCurrencyAmount" parent="ProcurementCardTransactionDetail-transactionOriginalCurrencyAmount-parentBean">
+        <property name="validationPattern">
+            <bean parent="FloatingPointValidationPattern" p:allowNegative="true" />
+        </property>
```
Comment (nataliac06) [inline kfs-core/src/main/resources/edu/arizona/kfs/fp/businessobject/datadictionary/ProcurementCardTransactionDetail.xml:112]: What is the proper way for validation for allowing negatives? I've seen this other variant in the kfs code:
<property name="validationPattern">
	<ref bean="FloatingPointAllowNegativeValidation" />
</property>

Also, are we missing a required='false' on ProcurementCardTransactionDetail-transactionSalesTaxAmount?
And I can't find the fix for: Enter Sales Tax Amount may allow negative amounts....

File: kfs-core/src/main/resources/edu/arizona/kfs/fp/businessobject/datadictionary/ProcurementCardHolder.xml (hunk 1 above)
Comment (adamkost) [reply kfs-core/src/main/resources/edu/arizona/kfs/fp/businessobject/datadictionary/ProcurementCardHolder.xml:210]: There is a maxLength in the parent's parent (PostalCode-parentBean), set to size 20, which should be sufficient for any incoming data (the spec says that there isn't supposed to be any validation on this field). The original mod had no change to the maxLength either.

File: kfs-core/src/main/resources/edu/arizona/kfs/fp/businessobject/datadictionary/ProcurementCardTransactionDetail.xml (hunk 2 above)
Comment (adamkost) [reply kfs-core/src/main/resources/edu/arizona/kfs/fp/businessobject/datadictionary/ProcurementCardTransactionDetail.xml:112]: transactionSalesTaxAmount is a required field, though the value may be 0.

When Enter Sales Tax Amount was implemented in UAF-2972, the settings to allow negative amounts are allowed by the base bean "GenericAttributes-genericAmount-parentBean" and the reference "GenericAmountValiationPattern"

**Review by nataliac06 — APPROVED** [review]
Looks good.

**Pull request merged**